<launch>
  <arg name="bag" default=""/>
  <arg name="seq" default="4_12"/>

  <rosparam command="load" file="$(find direct_stereo_slam)/cams/kitti/$(arg seq)/T_stereo.yaml" ns="direct_stereo_slam"/>
  <node name="direct_stereo_slam" pkg="direct_stereo_slam" type="slam_node" output="screen" required="true">
    <param name="topic0" value="/kitti/camera_gray_left/image_raw" />
    <param name="topic1" value="/kitti/camera_gray_right/image_raw" />
    <param name="calib0" value="$(find direct_stereo_slam)/cams/kitti/$(arg seq)/camera0.txt" />
    <param name="calib1" value="$(find direct_stereo_slam)/cams/kitti/$(arg seq)/camera1.txt" />
    <param name="bag" value="$(arg bag)" />
    <!-- <param name="scale_opt_thres" value="-1" /> -->
  </node>
  
    <!-- Launches ELAS node, and rectification nodes for input --> 
    <!-- Arguments: input stereo namespace and output elas namespace -->
    <arg name="stereo" default="kitti"/>
    <arg name="elas_ns" default="elas"/>
    
    <node name="$(arg elas_ns)" pkg="elas_ros" type="elas_ros" output="screen">
        <remap from="stereo" to="$(arg stereo)"/>
        <remap from="image" to="image_raw"/>

        <param name="disp_min" type="int" value="0"/>
        <param name="disp_max" type="int" value="255"/>
        <param name="support_threshold" type="double" value="0.95"/>
        <param name="support_texture" type="int" value="10"/>
        <param name="candidate_stepsize" type="int" value="5"/>
        <param name="incon_window_size" type="int" value="5"/>
        <param name="incon_threshold" type="int" value="5"/>
        <param name="incon_min_support" type="int" value="5"/>
        <param name="add_corners" type="bool" value="0"/>
        <param name="grid_size" type="int" value="20"/>
        <param name="beta" type="double" value="0.02"/>
        <param name="gamma" type="double" value="3"/>
        <param name="sigma" type="double" value="1"/>
        <param name="sradius" type="double" value="2"/>
        <param name="match_texture" type="int" value="1"/>
        <param name="lr_threshold" type="int" value="2"/>
        <param name="speckle_sim_threshold" type="double" value="1"/>
        <param name="speckle_size" type="int" value="200"/>
        <param name="ipol_gap_width" type="int" value="300"/>
        <param name="filter_median" type="bool" value="0"/>
        <param name="filter_adaptive_mean" type="bool" value="1"/>
        <param name="postprocess_only_left" type="bool" value="1"/>
        <param name="subsampling" type="bool" value="0"/>

        <param name="approximate_sync" value="true" />
        <param name="queue_size" type="int" value="5"/>
        <!-- If your cameras are not synchronised then uncomment the following line -->
        <!-- <param name="approximate_sync" value="true" type="bool"/> -->
    </node>

  </launch>
